### PR TITLE
feature: AQ::getOptimizedCalibrationCurve()

### PR DESCRIPTION
**Pull request by pcolaianni**
if it looks good, i'll sync with upstream and push it

### CODE REVIEW

File: src/openms/source/ANALYSIS/QUANTITATION/AbsoluteQuantitation.cpp [hunk 1]
```diff
@@ -636,10 +636,25 @@ namespace OpenMS
       }
       else
       {
-        LOG_INFO << "Warning: Standards not found for component " << quant_method.first << ".";
+        LOG_DEBUG << "Warning: Standards not found for component " << quant_method.first << ".";
```
Comment (dmccloskey) [inline src/openms/source/ANALYSIS/QUANTITATION/AbsoluteQuantitation.cpp:639]: Thanks!

File: src/openms/source/ANALYSIS/QUANTITATION/AbsoluteQuantitation.cpp [hunk 2]
```diff
@@ -636,10 +636,25 @@ namespace OpenMS
...
       }
     }
   } 
 
+  void AbsoluteQuantitation::getOptimizedCalibrationCurve(
+    const String& component_name,
+    std::vector<AbsoluteQuantitationStandards::featureConcentration>& component_concentrations
+  )
+  {
+    std::sort(component_concentrations.begin(), component_concentrations.end(),
```
Comment (dmccloskey) [inline src/openms/source/ANALYSIS/QUANTITATION/AbsoluteQuantitation.cpp]: Is this sort still needed?

File: src/openms/source/ANALYSIS/QUANTITATION/AbsoluteQuantitation.cpp [hunk 3]
```diff
@@ -636,10 +636,25 @@ namespace OpenMS
...
+      [](AbsoluteQuantitationStandards::featureConcentration lhs, AbsoluteQuantitationStandards::featureConcentration rhs)
+      {
+        return lhs.actual_concentration < rhs.actual_concentration; //ascending order
+      }
+    );
+    std::map<String, std::vector<AbsoluteQuantitationStandards::featureConcentration>> cc_map;
+    cc_map.insert({component_name, component_concentrations});
+    optimizeCalibrationCurves(cc_map);
```
Comment (dmccloskey) [inline src/openms/source/ANALYSIS/QUANTITATION/AbsoluteQuantitation.cpp]: Much cleaner.

File: src/pyOpenMS/pxds/AbsoluteQuantitation.pxd [hunk 4]
```diff
@@ -34,4 +34,7 @@ cdef extern from "<OpenMS/ANALYSIS/QUANTITATION/AbsoluteQuantitation.h>" namespa
         void optimizeCalibrationCurves(
            libcpp_map[ String, libcpp_vector[ AQS_featureConcentration ]] & components_concentrations) nogil except + # wrap-ignore
 
-
+        void getOptimizedCalibrationCurve(
```
Comment (dmccloskey) [inline src/pyOpenMS/pxds/AbsoluteQuantitation.pxd:37]: Can't wait to test this out.

File: src/tests/class_tests/openms/source/AbsoluteQuantitation_test.cpp [hunk 5]
```diff
@@ -707,6 +707,100 @@ START_SECTION((void optimizeCalibrationCurves(AbsoluteQuantitationStandards::com
 
 END_SECTION
 
+START_SECTION(void getOptimizedCalibrationCurve(
```
Comment (dmccloskey) [inline src/tests/class_tests/openms/source/AbsoluteQuantitation_test.cpp:710]: Looks good.

**Review by dmccloskey — CHANGES_REQUESTED** [review]
I just had one comment that needed to be addressed, but other than that I think it looks good.

File: src/openms/source/ANALYSIS/QUANTITATION/AbsoluteQuantitation.cpp (hunk 2 above)
Comment (pcolaianni) [reply src/openms/source/ANALYSIS/QUANTITATION/AbsoluteQuantitation.cpp]: no :+1:

**Comment by pcolaianni** [comment]
Thanks!

**Pull request closed without merging**